TAScheduler/UnitTests/tests.py

```diff
@@ -178,50 +189,778 @@ class TestCourseGetCrseInfo(TestCase):  # Randall
+            grader_status=True
+        )
+
+        tempuser2 = User(
+            email_address="test@instructor.com",
+            password="password",
+            first_name="firstin",
+            last_name="lastin",
+            home_address="Your mom's house",
+            phone_number=1234567890
+        )
+
+        tmpinstructor = Instructor.objects.create(
+            user=tempuser2
+        )
+
+        tmpcourse = Course.objects.create(
+            course_id=100,
+            semester="fall 2023",
+            name="testCourse",
+            description="test",
+            num_of_sections=3,
+            modality="online",
+            credits=3
+        )
+
+        tmpsection = Section.objects.create(
+            section_id=1011,
+            course=tmpcourse,
+            location="Cool place",
+            meeting_time=datetime.datetime
+        )
+
+        tmplec = Lecture.objects.create(
+            section=tmpsection,
+            ta=tmpta,
+            instructor=tmpinstructor
+        )
+
+        self.lecture = LectureObj(tmplec)
+
+    def test_lecture_make(self):
+        self.assertIsNotNone(self.lecture, "__init__ failed in making Lecture")
+
+    def test_bad_lecture_make(self):
+        with self.assertRaises(TypeError, msg="__init__ failed in making Lecture: Type error"):
+            self.lecture = LectureObj(7)
+
+
+class TestLectureGetLecInstrAsgmt(unittest.TestCase):  # Joe
+    instructor = None
+    lecture = None
+
+    def setUp(self):
+        tempuser = User(
+            email_address="test@ta.com",
+            password="password",
+            first_name="first",
+            last_name="last",
+            home_address="Your mom's house",
+            phone_number=1234567890
+        )
+
+        tmpta = TA.objects.create(
+            user=tempuser,
+            grader_status=True
+        )
+
+        tmpcourse = Course.objects.create(
+            course_id=100,
+            semester="fall 2023",
+            name="testCourse",
+            description="test",
+            num_of_sections=3,
+            modality="online",
+            credits=3
+        )
+
+        tmpsection = Section.objects.create(
+            section_id=1011,
+            course=tmpcourse,
+            location="Cool place",
+            meeting_time=datetime.datetime
+        )
+
+        tmplec = Lecture.objects.create(
+            section=tmpsection,
+            ta=tmpta,
+        )
+
+        tempuser2 = User(
+            email_address="test@instructor.com",
+            password="password",
+            first_name="firstin",
+            last_name="lastin",
+            home_address="Your mom's house",
+            phone_number=1234567890
+        )
+
+        self.instructor = Instructor.objects.create(
+            user=tempuser2
+        )
+        self.instructor.save()
+
+        self.lecture = LectureObj(tmplec)
+
+    def test_get_instructor(self):
+        self.lecture.addInstr(self.instructor)
```

Comment:
line 599 defines lecture object with the tmpLec DB item ... that DB item (line 580) already has a TA assigned to it ... a section is only allowed 1 user per section ... 

Solution: instead of using your self.lecture, create a new lecture object with an empty user assignment assignment

Reply:
This is the get instructor tests I believe that you think it is the get ta tests? Just in case I'll remove the TA as well since it won't be needed anyways